tests/test_harvesters.py

```diff
@@ -35,6 +46,19 @@ def test_exception_on_base_harvester_create_ingester(self):
             with self.assertRaises(NotImplementedError):
                 base_harvester._create_ingester()
 
+    def test_exception_on_incorrect_ingester_creation(self):
+        """
+        An exception is raised if an attempt is made to call the _create_ingester() method of the
+        base Harvester class with incorrect configuration.
+        In this test 'max_fetcher_threads' is misspelled with 'max_fetcher_thread'
+        """
+        with mock.patch.object(harvesters.Harvester, '__init__', return_value=None):
+            self.config = {'max_db_threads': 1, 'max_fetcher_thread': 2}
+            base_harvester = harvesters.WebDirectoryHarvester()
+            base_harvester.ingester = ingesters.DDXIngester
+            with self.assertRaises(HarvesterConfigurationError):
+                base_harvester._create_ingester()
```

Comment:
This test does not do what you think:

- `self` refers to the `HarvesterTestCase` instance, so you set a config attribute for the TestCase instance, not for `base_harvester`

- The test passes because `base_harvester` does not have a `config` attribute (because you mocked the constructor), which raises an `AttributeError` in `_create_ingester`, which is caught and then raises a `HarvesterConfigurationError`.
So the `HarvesterConfigurationError` is not raised because of the misspelling in the configuration.

- Then, and maybe more importantly, there is no code in the harvester to make sure that 'max_fetcher_threads' and 'max_db_threads' are present in the configuration, so this test should not exist.

- All this makes me doubt that catching `AttributeError` in `_create_ingester` and `_create_crawlers` is necessary. Unless there is a good reason it should be removed to avoid this kind of situation where the real behavior is hidden.

Reply:
Done in the next commit